app/navigation/Navigator.scala

```diff
@@ -245,6 +245,19 @@ class Navigator @Inject() (categorisationService: CategorisationService) {
 
   private def navigateFromCategorisationPreparationPage(answers: UserAnswers, recordId: String): Call =
     answers.get(CategorisationDetailsQuery(recordId)) match {
+      case Some(catInfo)
+          if !catInfo.isNiphlAuthorised && catInfo.categoryAssessments.exists(ass => ass.isNiphlsAnswer) =>
```

Comment:
would be nice to add to the categoryInfo a function that is like
def isNiphlsAssessment = categoryAssessments.exists(ass => ass.isNiphlsAnswer) && categoryAssessments.exists(ass => ass.isCategory2 && ass.hasNoAnswers), and call that here